README.md

```diff
@@ -15,95 +17,254 @@ with the cloudiness of [AWS S3 buckets](https://aws.amazon.com/s3/)
+This container also supports passing sensitive values via [Docker
+secrets](https://docs.docker.com/engine/swarm/secrets/).  Passing sensitive
+values like your credentials can be more secure using secrets than using
+environment variables.  See the
+[secrets](#secrets) section below for a table of all supported secret files.
+
+1. To use secrets, create a `certboto_credentials` file containing the values you
+want set:
+
+    ```ini
+    [default]
+    aws_access_key_id = XXXXXXXXXXXXXXXXXXXX
+    aws_secret_access_key = xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+
+    [dns-role]
+    role_arn = arn:aws:iam::1234567890ab:role/ModifyPublicDNS
+    source_profile = default
+
+    [bucket-role]
+    role_arn = arn:aws:iam::1234567890ab:role/CertbotBucket
+    source_profile = default
+
+    # If running on EC2 with an instance profile that allows sts:AssumeRole
+    # you can assume delegated roles using the metadata as the credential source
+    # See: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-role.html
+
+    [dns-role-ec2]
+    role_arn = arn:aws:iam::1234567890ab:role/ModifyPublicDNS
+    credential_source = Ec2InstanceMetadata
 
-Create a `docker-compose.yml` file similar to this:
+    [bucket-role-ec2]
+    role_arn = arn:aws:iam::1234567890ab:role/CertbotBucket
+    credential_source = Ec2InstanceMetadata
+    ```
 
-```yml
----
-version: "3.7"
+1. Then add the secret to your `docker-compose.yml` file:
 
-secrets:
-  credentials:
-    file: /home/username/.aws/credentials
+    ```yaml
+    ---
+    version: "3.7"
 
-services:
-  certboto:
-    image: cisagov/certboto
-    init: true
-    restart: "no"
-    environment:
-      - AWS_DEFAULT_REGION=us-east-1
-      - BUCKET_NAME=my-certificates
-      - BUCKET_PROFILE=certsync-role
-      - DNS_PROFILE=dns-role
     secrets:
-      - source: credentials
-        target: credentials
-```
+      credentials:
+        file: certboto_credentials
 
-Pull `cisagov/certboto` from [Docker hub](https://hub.docker.com):
+    services:
+      certboto:
+        image: cisagov/certboto
+        init: true
+        restart: "no"
+        environment:
+          - AWS_DEFAULT_REGION=us-east-1
+          - BUCKET_NAME=my-certificates
+          - BUCKET_PROFILE=certsync-role
+          - DNS_PROFILE=dns-role
+        secrets:
+          - source: credentials
+            target: credentials
+    ```
 
-```console
-docker-compose pull
-```
+## Updating your container ##
+
+### Docker Compose ###
+
+1. Pull the new image from Docker Hub:
+
+    ```console
+    docker-compose pull
+    ```
+
+1. Recreate the running container by following the
+[previous instructions](#running-with-docker-compose):
+
+    ```console
+    docker-compose run certboto
+    ```
+
+## Image tags ##
+
+The images of this container are tagged with [semantic
+versions](https://semver.org) of the underlying example project that they
+containerize.  It is recommended that most users use a version tag (e.g.
+`:0.0.3`).
```

Comment:
I think pinning the [certbot/dns-route53](https://hub.docker.com/r/certbot/dns-route53) is a good strategy. Docker images should pin everything installed if possible.

I am unsure about the dual versioning approach as it is counter to typical versioning guidelines because the `cisa.x` will not reset when there is an increment in the [certbot/dns-route53](https://hub.docker.com/r/certbot/dns-route53) version. We could alternatively increment our own version and adjust the tag to include the [certbot/dns-route53](https://hub.docker.com/r/certbot/dns-route53) version being used, but I'm not sure if that's any better. Something like `v0.0.3(1.18.0)`, but again I'm not sure that's any better.

Reply:
I was thinking 

> I am unsure about the dual versioning approach as it is counter to typical versioning guidelines because the `cisa.x` will not reset when there is an increment in the [certbot/dns-route53](https://hub.docker.com/r/certbot/dns-route53) version.

Let's say we are pinned to version `certbot/dns-route53:v1.18.0` and our version is `cisagov/certboto:1.18.0-cisa.3`.  Then, when `certbot/dns-route53:v1.19.0` comes out, we would update our pin to that version and release `cisagov/certboto:1.19.0-cisa.1`.  Until we update that pin, we are free to release `cisagov/certboto:1.18.0-cisa.4` and so on.  I agree that it violates semantic versioning rules as far as our repo goes since we only have the `cisa.x` value to increment, but I'm not sure how much it matters in repos like this where most of the functionality lives in the base image (`certbot/dns-route53`).

Also, `cisagov/certboto:1.18.0-cisa.3` is compliant with semantic versioning ([as a pre-release](https://semver.org/#spec-item-9)), whereas something like `v0.0.3(1.18.0)` is not.  FYI, I previously tried to use the [metadata indicator (`+` sign)](https://semver.org/#spec-item-10) instead of the hyphen in our version string, but that broke something in our workflow (I don't remember exactly what).

So maybe we should combine both of our ideas into something like `cisagov/certboto:1.18.0-cisa.0.0.3`?

Reply:
As I recall, Docker doesn't allow a plus character in the version string.  I'm guessing that is what breaks in the workflow.  See also, for example, [here](https://github.com/cisagov/skeleton-docker/blob/develop/.github/workflows/build.yml#L206) to see how we deal with this problem in another context.

Reply:
> As I recall, Docker doesn't allow a plus character in the version string. I'm guessing that is what breaks in the workflow. 

Yes, that sounds familiar now that you mention it.

Reply:
So after thinking about it more I do not think we should use the [certbot/dns-route53](https://hub.docker.com/r/certbot/dns-route53) version as the tag for this project. That Docker image is just a dependency, whereas I believe the version number should only be mirrored by the Docker image when you are Dockerizing a tool. In the Gophish examples you mentioned they are both the Dockerization of the Gophish tool. What we are versioning against is the functionality to use an AWS S3 bucket to manage certificates created using `certbot` and not the `certbot` tool itself.

Thank you for coming to my TED talk.

Reply:
That is an excellent distinction and I now agree with you.  

Though I still think we should pin this image to a specific version of [certbot/dns-route53](https://hub.docker.com/r/certbot/dns-route53).

Reply:
Also, don't forget that you still need to update the "example" verbiage that spawned this thread.

Reply:
@dav3r Please see a838bd7 for the README update and #26 to track pinning the image's dependencies (including the [certbot/dns-route53](https://hub.docker.com/r/certbot/dns-route53) image used).

Reply:
This all looks great to me - thanks!